packages/web-components/src/components/ic-badge/ic-badge.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  Component,
+  Element,
+  Host,
+  Method,
+  Prop,
+  State,
+  h,
+} from "@stencil/core";
+import {
+  IcBadgePositions,
+  IcBadgeTypes,
+  IcBadgeVariants,
+} from "./ic-badge.types";
+import {
+  IcColorRGBA,
+  IcSizes,
+  IcThemeForeground,
+  IcThemeForegroundEnum,
+} from "../../utils/types";
+import {
+  getCssProperty,
+  getParentElement,
+  hexToRgba,
+  rgbaStrToObj,
+} from "../../utils/helpers";
+
+@Component({
+  tag: "ic-badge",
+  styleUrl: "ic-badge.css",
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class Badge {
+  private customColorRGBA: IcColorRGBA;
+  private foregroundColour: IcThemeForeground;
+
+  @Element() el: HTMLIcBadgeElement;
+
+  @State() visible: boolean = true;
+
+  /**
+   * The accessible label of the badge component to provide context for screen reader users.
+   */
+  @Prop() accessibleLabel?: string;
+
+  /**
+   * The custom badge colour. This will only style the badge component if variant="custom".
+   * Can be a hex value e.g. "#ff0000", RGB e.g. "rgb(255, 0, 0)", or RGBA e.g. "rgba(255, 0, 0, 1)".
+   */
+  @Prop() customColor?: string = null;
+
+  /**
+   * The maximum number shown on your badge appended with a +.
+   * This will only be displayed if type="text" and textLabel is not empty.
+   */
+  @Prop() maxNumber?: number;
+
+  /**
+   * The positioning of the badge in reference to the parent element.
+   */
+  @Prop() position?: IcBadgePositions = "far";
+
+  /**
+   * The size of the badge to be displayed.
+   */
+  @Prop() size?: IcSizes = "default";
+
+  /**
+   * The text displayed in the badge. This will only be displayed if type="text".
+   */
+  @Prop() textLabel?: number;
```

Comment:
Not sure, the AC was:
**Given that a text badge is being implemented
When setting it properties
Then the text label can be set to any number.**
I took that to be that it could only be set to a number as there was no ACs around other text.
Might be worth @DBD324 weighing in on this.

Reply:
Good question, I think the primary use case is for numbers, but can see how something like new could be useful too. We'll include in guidance detail about avoiding long labels. Maybe we should keep it flexible and allow text to be entered too. Would you suggest keeping its type 'string' with numbers being able to be added, or would we need to allow the change of type from number to string?

Reply:
I think change the type of the prop to 'string' as that will allow numbers and letters (I think it would bring up an error otherwise when trying to pass in a letter). Then in the code it can be converted to a number where necessary - to carry out calculations for the max number etc. (if the value of the string is actually a number)